src/OrleansLoggerProvider/LegacyOrleansLogger.cs

```diff
@@ -0,0 +1,149 @@
+        private readonly string name;
+        private readonly IList<IFlushableLogConsumer> flushableConsumers;
+        private readonly LoggerType loggerType;
+        private readonly IPEndPoint ipEndPoint;
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="categoryName"></param>
+        /// <param name="logConsumers"></param>
+        /// <param name="maxSeverityLevel"></param>
+        public LegacyOrleansLogger(string categoryName, IList<ILogConsumer> logConsumers, IPEndPoint ipEndPoint)
+        {
+            this.logConsumers = logConsumers;
+            this.flushableConsumers = logConsumers.OfType<IFlushableLogConsumer>().ToList();
+            this.name = categoryName;
+            this.loggerType = DetermineLoggerType(categoryName);
+            this.ipEndPoint = ipEndPoint;
+        }
+
+        /// <inheritdoc/>
+        public IDisposable BeginScope<TState>(TState state)
+        {
+            if (state == null)
+            {
+                throw new ArgumentNullException(nameof(state));
+            }
+            //TODO: support logging scope
+            return NullScope.Instance;
+        }
+
+        /// <inheritdoc/>
+        public bool IsEnabled(LogLevel logLevel)
+        {
+            return true;
+        }
+
+        /// <summary>
+        /// Log a message. Current logger supports legacy event bulking feature. Message bulking feature will only log eventId code appearance count
+        /// if certain event appear more than <see cref="EventBulkingConfig.BulkEventLimit"/>> in <see cref="EventBulkingConfig.BulkEventInterval"/>
+        /// </summary>
+        /// <typeparam name="TState"></typeparam>
+        /// <param name="logLevel"></param>
+        /// <param name="eventId"></param>
+        /// <param name="state"></param>
+        /// <param name="exception"></param>
+        /// <param name="formatter"></param>
+        public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
+        {
+            var errorCode = eventId.Id;
+            var severity = LogLevelToSeverity(logLevel);
+            WriteLogMessageToLogConsumers(errorCode, severity, formatter(state, exception), exception);
+        }
+
+        /// <summary>
+        /// Map LogLevel to Severity
+        /// </summary>
+        /// <param name="logLevel"></param>
+        /// <returns></returns>
+        public static Severity LogLevelToSeverity(LogLevel logLevel)
+        {
+            switch (logLevel)
+            {
+                case LogLevel.None: return Severity.Off;
+                case LogLevel.Critical: return Severity.Error;
+                case LogLevel.Error: return Severity.Error;
+                case LogLevel.Warning: return Severity.Warning;
+                case LogLevel.Information: return Severity.Info;
+                case LogLevel.Debug: return Severity.Verbose;
```

Comment:
Since we cannot map ```Verbose2``` due to the lack of "enough" levels in the logging abstractions, IMHO we should replace all ```Verbose2``` calls with ```Verbose``` or ```Verbose3```.

For default I'd add at least a comment that it for ```Trace```, or explicitly make that a case (I prefer), and for default throw an exception (since that cannot happen, because we're mapping all values).

Reply:
I believe this is not part of this PR's scope? and we are not mapping Verbose2 and Verbose3 to Trace, instead of "cannot map Verbose2"

Reply:
it map LogLevel.Trace to Verbose3 here so that both Verbose2 and Verbose3 will be logged if user set default llogLevel to Trace.